### PR TITLE
Bluetooth: Controller: df: Fix lack of IQ reports when IQ sampling enabled and received PDU includes CTE

### PR DESCRIPTION
**Pull request by ppryga-nordic**
The radio_df_cte_ready function returns state of EVENTS_CTEPRESENT.
The function may not be used in create_iq_report function, because
it is called after lll_isr_rx_status_reset. The lll_isr_rx_status-
_reset clears EVENTS_CTEPRESENT value.

The result of this calls chain is lack of IQ sample report no matter
if the PDU has CTEInfo and radio has collected IQ samples.

To fix the problem, state of EVENTS_CTEPRESENT has to be get when
e.g. rssi_ready is get.

Second thing that is change by this PR is related with a EVENTS_CTEPRESENT.
Clear of the event was compiled only for DF connected mode.
It should be compiled in for both connectionless and connected mode.

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c [hunk 1]
```diff
@@ -52,7 +52,7 @@ static int prepare_cb_common(struct lll_prepare_param *p, uint8_t chan_idx);
 static int is_abort_cb(void *next, void *curr, lll_prepare_cb_t *resume_cb);
 static void abort_cb(struct lll_prepare_param *prepare_param, void *param);
 static int isr_rx(struct lll_sync *lll, uint8_t node_type, uint8_t crc_ok, uint8_t rssi_ready,
-		  enum sync_status status);
+		  enum sync_status status, uint8_t cte_ready);
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c:55]: Again my preference is to have the `cte_ready` before `status` or before `rssi_ready`, as it is radio related.

**Review by Thalley — DISMISSED** [review]

**Review by cvinayak — DISMISSED** [review]
Minor suggestions, otherwise looks good.

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c (hunk 1 above)
Comment (ppryga) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c]: I've moved `cte_ready` before `rssi_ready`.

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c [hunk 2]
```diff
@@ -730,8 +741,14 @@ static void isr_rx_adv_sync_estab(void *param)
 		rssi_ready = radio_rssi_is_ready();
 		sync_ok = sync_filtrate_by_cte_type(lll->cte_type, lll->filter_policy);
 		trx_cnt = 1U;
+
+#if defined(CONFIG_BT_CTLR_DF_SCAN_CTE_RX)
```
[the remark this answers is not in the document]
Comment (ppryga) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c]: Changed to use `IS_EANBLED`

**Review by cvinayak — DISMISSED** [review]

**Review by Thalley — DISMISSED** [review]

**Comment by ppryga** [comment]
I've added one new commit that removes `#if defined(CONFIG_BT_CTLR_DF)` from `subsys/bluetooth/controller/hal/radio_df.h`.
That prevented `subsys/bluetooth/controller/ll_sw/nordic/hal/radio/radio_df.h` from being included. As a result of that compiler was complaining about implicit declaration of `radio_df_cte_ready()`.

**Review by cvinayak — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Pull request merged**